app/models/task/TaskType.scala

```diff
@@ -120,80 +113,73 @@ class TaskTypeDAO @Inject()(sqlClient: SqlClient)(implicit ec: ExecutionContext)
-                       """)
+                    values(${t._id}, $organizationId, ${t._team}, ${t.summary}, ${t.description},
+                           ${EnumerationArrayValue(t.settings.allowedModes, "webknossos.TASKTYPE_MODES")},
+                           ${t.settings.preferredMode},
+                           ${t.settings.branchPointsAllowed},
+                           ${t.settings.somaClickingAllowed},
+                           ${t.settings.volumeInterpolationAllowed},
+                           ${t.settings.mergerMode},
+                           ${t.settings.resolutionRestrictions.min},
+                           ${t.settings.resolutionRestrictions.max},
+                           ${t.recommendedConfiguration.map(Json.toJson(_))},
+                           ${t.tracingType},
+                           ${t.created}, ${t.isDeleted})
+                    """.asUpdate)
     } yield ()
 
   def updateOne(t: TaskType)(implicit ctx: DBAccessContext): Fox[Unit] =
     for { // note that t.created is immutable, hence skipped here
       _ <- assertUpdateAccess(t._id)
-      allowedModesLiteral = sanitize(writeArrayTuple(t.settings.allowedModes.map(_.toString)))
-      resolutionMinLiteral = optionLiteral(t.settings.resolutionRestrictions.min.map(_.toString))
-      resolutionMaxLiteral = optionLiteral(t.settings.resolutionRestrictions.max.map(_.toString))
-      configurationLiteral = optionLiteral(t.recommendedConfiguration.map(c => sanitize(Json.toJson(c).toString)))
-      _ <- run(sqlu"""update webknossos.taskTypes
-                        set
-                         _team = ${t._team.id},
-                         summary = ${t.summary},
-                         description = ${t.description},
-                         settings_allowedModes = '#$allowedModesLiteral',
-                         settings_preferredMode = #${optionLiteral(t.settings.preferredMode.map(sanitize))},
-                         settings_branchPointsAllowed = ${t.settings.branchPointsAllowed},
-                         settings_somaClickingAllowed = ${t.settings.somaClickingAllowed},
-                         settings_volumeInterpolationAllowed = ${t.settings.volumeInterpolationAllowed},
-                         settings_mergerMode = ${t.settings.mergerMode},
-                         settings_resolutionRestrictions_min = #$resolutionMinLiteral,
-                           settings_resolutionRestrictions_max = #$resolutionMaxLiteral,
-                           recommendedConfiguration = #$configurationLiteral,
-                           isDeleted = ${t.isDeleted}
-                          where _id = ${t._id.id}""")
+      _ <- run(q"""UPDATE webknossos.taskTypes
+                   SET
+                   _team = ${t._team},
+                   summary = ${t.summary},
+                   description = ${t.description},
+                   settings_allowedModes = ${EnumerationArrayValue(t.settings.allowedModes, "webknossos.TASKTYPE_MODES")},
```

Comment:
Would it be possible to add an sqlTypeName attribute to the enumeration classes? Or make dedicated *toSqlValue functions? I would prefer to not have this boilerplate in the queries.

Reply:
I tried that but the only solution I could figure out involved [this](https://stackoverflow.com/questions/44390775/scala-reflection-get-class-of-enumaration-value) private method to access the Enum class from the value object. I’d say that is not worth it at this time. I’d defer finding a better solution here, and since it task type allowed modes is the only occurence of enum-arrays in postgres, let’s keep it like this for the moment.